optima/optimization.py

```diff
@@ -126,16 +176,69 @@ def outcomecalc(budgetvec=None, project=None, parset=None, progset=None, objecti
         thisoutcome = results.main['num'+key].tot[0][indices].sum() # the instantaneous outcome e.g. objectives['numdeath'] -- 0 is since best
         outcome += thisoutcome*thisweight*results.dt # Calculate objective
 
+    # Optionally start debugger
+    if debug: import traceback; traceback.print_exc(); import pdb; pdb.set_trace()
+
     if outputresults:
         results.outcome = outcome
         results.budgetvec = budgetvec # WARNING, not sure this should be here
         results.budgetyears = [objectives['start']] # WARNING, this is ugly, should be made less kludgy
-        results.budget = vec2budget(progset, budgetvec) # Convert to budget
+        results.budget = vec2budget(progset, budgetvec, optiminds) # Convert to budget
         return results
     else: 
         return outcome
 
 
+def constrainbudget(origbudget, total=None, limits=None, eps=1e-3):
```

Comment:
it's a different eps since it's about a tolerance for a fit rather than to prevent divide-by-zero, plus settings isn't available here so i thought it might introduce an annoying and not really necessary dependency